File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int index=0;

        for (int i=0; i<nums.length; i++) {
            if (index==i && nums[i]==0)
                break;
            index=Math.max(index, i+nums[i]);
        }

        if (index>=nums.length-1)
            return true;
        else
            return false;
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums)==1:
            return True
        queue = [(0, nums[0])]
        n = len(nums)

        while queue:
            index, steps = queue.pop()
            
            for i in range(1, steps+1):
                if index+i>=n-1:
                    return True
                elif nums[index+i]!=0:
                    queue.append((index+i, nums[index+i]))

        return False